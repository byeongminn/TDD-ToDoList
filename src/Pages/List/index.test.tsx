/*
  <List 페이지 컴포넌트 사양>
  1. 할 일 목록을 보여준다. (할 일 목록은 localStorage에 저장되어 있다.)
  2. 할 일 목록은 할 일 뿐만 아니라 해당 할 일을 삭제할 수 있는 삭제 버튼을 가지고 있다.
  3. 각각의 할 일 목록을 클릭하면 할 일 상세페이지('/detail/:id')로 이동한다.
  4. 추가 버튼('+')을 가지고 있다.
  5. 추가 버튼을 누르면 할 일 추가 페이지('/add')로 이동한다.
*/

import { render, screen } from '@testing-library/react';

describe('<List />', () => {
  it('renders component correctly', () => {
    localStorage.setItem('ToDoList', '["ToDo 1", "ToDo 2", "ToDo 3"]');

    render(<List />);

    const toDoItem1 = screen.getByText('ToDo 1');
    expect(toDoItem1).toBeInTheDocument();

    const toDoItem2 = screen.getByText('ToDo 2');
    expect(toDoItem2).toBeInTheDocument();

    const toDoItem3 = screen.getByText('ToDo 3');
    expect(toDoItem3).toBeInTheDocument();
  });
});
