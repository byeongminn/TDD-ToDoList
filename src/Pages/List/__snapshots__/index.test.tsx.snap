// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders component correctly 1`] = `
<div>
  <div
    class="sc-iGgVNO jvnxrv"
  >
    <div
      class="sc-gsFSjX exaqRz"
    >
      <div
        class="sc-kAycRU dCZHbL"
      >
        <a
          class="sc-imWZod fbRJDM"
          href="/detail/0"
        >
          ToDo 1
        </a>
        <div
          class="sc-fqkwJk BIzEt"
        >
          <div
            class="sc-dcJtft drInyG"
          >
            삭제
          </div>
        </div>
      </div>
      <div
        class="sc-kAycRU dCZHbL"
      >
        <a
          class="sc-imWZod fbRJDM"
          href="/detail/1"
        >
          ToDo 2
        </a>
        <div
          class="sc-fqkwJk BIzEt"
        >
          <div
            class="sc-dcJtft drInyG"
          >
            삭제
          </div>
        </div>
      </div>
      <div
        class="sc-kAycRU dCZHbL"
      >
        <a
          class="sc-imWZod fbRJDM"
          href="/detail/2"
        >
          ToDo 3
        </a>
        <div
          class="sc-fqkwJk BIzEt"
        >
          <div
            class="sc-dcJtft drInyG"
          >
            삭제
          </div>
        </div>
      </div>
    </div>
    <a
      class="sc-jXbVAB eSbMMU"
      href="/add"
    >
      +
    </a>
  </div>
</div>
`;
